### ISSUE TITLE
Ability to control soap-env and ns

### ISSUE THREAD

**Issue by orzilca** [mvantellingen/python-zeep#60]
I was trying to implement a wsdl request an encountered an issue.

the accpected request should be like:
```
<?xml version="1.0" encoding="utf-8"?>
<soap:Envelope xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
<soap:Header>
<AuthenticateRequest soap:mustUnderstand="1" xmlns="http://apitest.trafficvance.com/?v3=campaigns">
<apiKey xmlns="">apikey</apiKey>
</AuthenticateRequest>
</soap:Header>
<soap:Body>
<addCampaign xmlns="http://apitest.trafficvance.com/?v3=campaigns">
<parameters xmlns="">
<campaign>
<name>Test Campaign</name>
<description>This is test</description>
<dailyBudgetSunday>0</dailyBudgetSunday>
<dailyBudgetMonday>0</dailyBudgetMonday>
<dailyBudgetTuesday>0</dailyBudgetTuesday>
<dailyBudgetWednesday>0</dailyBudgetWednesday>
<dailyBudgetThursday>0</dailyBudgetThursday>
<dailyBudgetFriday>0</dailyBudgetFriday>
<dailyBudgetSaturday>0</dailyBudgetSaturday>
</campaign>
</parameters>
</addCampaign>
</soap:Body>
</soap:Envelope>
```

The request message issued by Zeep looks like:
```
<soap-env:Envelope xmlns:ns0="http://apitest.trafficvance.com/?v3=campaigns" xmlns:soap-env="http://schemas.xmlsoap.org/soap/envelope/">
<soap-env:Header>
<ns0:AuthenticateRequest>
<apiKey>apikey</apiKey>
</ns0:AuthenticateRequest>
</soap-env:Header>
<soap-env:Body>
<ns0:addCampaignRequest>
<parameters>
<campaign>
<name>Test Campaign</name>
<description>This is test</description>
<dailyBudgetSunday>0</dailyBudgetSunday>
<dailyBudgetMonday>0</dailyBudgetMonday>
<dailyBudgetTuesday>0</dailyBudgetTuesday>
<dailyBudgetWednesday>0</dailyBudgetWednesday>
<dailyBudgetThursday>0</dailyBudgetThursday>
<dailyBudgetFriday>0</dailyBudgetFriday>
<dailyBudgetSaturday>0</dailyBudgetSaturday>
</campaign>
</parameters>
</ns0:addCampaignRequest>
</soap-env:Body>
</soap-env:Envelope>
```

There are a few issues here:
- ```<soap-env:...``` instead of ```<soap:...```
- extra unnecessary ```ns0:```
- missing ```<?xml version="1.0" encoding="utf-8"?>```
- missing ```xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"``` and ```xmlns:xsd="http://www.w3.org/2001/XMLSchema"```

I've looked at the entire source code trying to see if these are controllable, couldn't find anything..

**Comment by mvantellingen** [comment]
Hiya, thanks for the complete 

Initial feedback on your reported issues:
- namespace prefix can differ and should have this should have no impact.
- The ``ns0`` namespaces is actually used in the xml. The xml is pulled through ``lxml.etree.cleanup_namespaces()``
- I've just added the missing xml declaration
- The namespaces are not used? (and correctly removed by lxml cleanup_namespaces)

The problem might be that we are missing the ``soap:mustUnderstand="1"`` attribute

**Comment by orzilca** [comment]
I think there is another issue here.

I'm calling addCampaign service, and it expects to get an addCampaign element:
```<addCampaign xmlns="http://apitest.trafficvance.com/?v3=campaigns">```

But instead, zeep generates:
<ns0:addCampaignRequest>

---

This is the code i'm using:
```
    transport = Transport(verify=False)
    client = Client(
        'http://apitest.trafficvance.com/?v3=' + service + '.wsdl',
        transport=transport
    )

    request_type = client.get_type('ns0:CampaignAdd')
    request = request_type(
        name='Test Campaign',
        description='This is test',
        dailyBudgetSunday=0,
        dailyBudgetMonday=0,
        dailyBudgetTuesday=0,
        dailyBudgetWednesday=0,
        dailyBudgetThursday=0,
        dailyBudgetFriday=0,
        dailyBudgetSaturday=0
    )

    _response = client.service.addCampaign(
        parameters={
            'campaign': request
        },
        _soapheader={'apiKey': 'some_key'}
    )
```

**Comment by mvantellingen** [comment]
Do you have a wsdl somewhere?

**Comment by orzilca** [comment]
yes,
http://apitest.trafficvance.com/?v3=campaigns.wsdl

**Comment by mvantellingen** [comment]
I just fixed the rpc message serializer so that addCampaignRequest is now addCampaign. The issue was that for RPC messages the operation name instead of the message name should be used.

I haven't tested the RPC message style much since most services use the Document message style, so let me know if you run into more issues

**Comment by orzilca** [comment]
Awesome!
When will you release the changes?

**Comment by mvantellingen** [comment]
If this works I suppose I can release a 0.8.1. Can you check? ``pip install https://github.com/mvantellingen/python-zeep/archive/master.zip#egg=zeep``

**Comment by orzilca** [comment]
works !

**Comment by mvantellingen** [comment]
Just released 0.8.1 to pypi.
